http/cookie.ts

```diff
@@ -1,11 +1,103 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-import { ServerRequest } from "./server.ts";
+import { ServerRequest, Response } from "./server.ts";
+import { assert } from "../testing/asserts.ts";
+import { pad } from "../strings/pad.ts";
 
 export interface Cookie {
   [key: string]: string;
 }
 
-/* Parse the cookie of the Server Request */
+export interface CookieValue {
+  name: string;
+  value: string;
+}
+
+export interface CookieOptions {
+  Expires?: Date;
+  MaxAge?: number;
+  Domain?: string;
+  Path?: string;
+  Secure?: boolean;
+  HttpOnly?: boolean;
+  SameSite?: SameSite;
+}
```

Comment:
Use lower-case for the first letter.

It would be nice to have an element like this ([as is done in Go](https://golang.org/pkg/net/http/#Cookie))
```
        unparsed string[] // Raw text of unparsed attribute-value pairs
```

Reply:
Refactored and added tests.